content/blogs/2024-04-05-getting-started-with-kestra.md

```diff
@@ -0,0 +1,315 @@
+
+```bash
+docker run --pull=always --rm -it -p 8080:8080 --user=root \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  -v /tmp:/tmp kestra/kestra:latest-full server local 
+```
+
+Once you’ve run this command, head over to your browser and open [https://localhost:8080](https://localhost:8080) to launch the interface so we can start building workflows!
+
+## Properties
+
+Before we start making our workflows, it’s worth understand a few fundamental properties that we’ll use to build everything!
+
+Workflows are referenced as Flows and they’re declared using YAML making it very readable as well as works with any language! Within each flow, there are 3 required properties you’ll need:
+
+- `id` which is the name of your flow. This can’t be changed once you’ve executed your flow for the first time.
+- `namespace` which allows you to specify the environments you want your flow to execute in, e.g. production vs development
+- `tasks` which is a list of the tasks that will execute when the flow is executed, in the order they’re defined in. Tasks contain an `id` as well as a `type` with each different type having their own additional properties.
+
+To help visualise it, here's an example:
+
+```yaml
+id: getting_started
+namespace: example
+tasks:
+  - id: hello_world
+    type: io.kestra.core.tasks.log.Log
+    message: Hello World!
+```
+
+Everything builds off of these 3 properties but there’s a few more optional properties that you’ll want to use to give you full flexibility.
+
+- **Inputs:** Instead of hardcoding values into your flows, you can set them as constant values separately. Great if you plan to reuse them in multiple tasks. An input might look like this: `{{ inputs.webhook_url }}` .
+- **Outputs:** Tasks will often generate outputs that you’ll want to pass on to a later task. Outputs let you connect both variables as well as files to later tasks. An output of a variable could look like this: `"{{ outputs.script.vars['output'] }}"`
+- **Triggers:** Instead of manually executing your flow, you can setup triggers to execute it based on a set of conditions such as time schedule or a webhook.
+
+The last thing to mention is Plugins! To help you build powerful flows, you can utilise Plugins for tools and platforms you already use to help speed things up! Every plugin is different but we'll cover a few examples later in the blog.
+
+While that might be a lot of properties to get your head around, the Kestra platforms interactive topology will help us configure these correctly!
+
+## Building our First Flow
+
+For our first flow, we're going to setup a simple automation that runs a Python script once every hour and sends its output to Discord as a notification! Let's start with the Python part! Firstly, we need a Python file for Kestra to execute! We’ll use something really simple that generates an output from an API request. 
```

Comment:
```suggestion
For our first flow, we're going to set up a simple automation that runs a Python script once every hour and sends its output to Discord as a notification. Let's start with the Python part. Firstly, we need a Python file for Kestra to execute! We’ll use something really simple that generates an output from an API request. 
```

I noticed throughout the post that there are many exclamation marks. Consider removing it if it gets too much (e.g. here, it feels not needed)